src/overlays/actors/ovl_En_Crow/z_en_crow.c

```diff
@@ -21,41 +30,446 @@ const ActorInit En_Crow_InitVars = {
+    this->actionFunc = func_809E1004;
+}
+
+void func_809E0770(EnCrow* this) {
+    if (sDeathCount == 10) {
+        this->actor.params = 1;
+        sDeathCount = 0;
+        this->collider.list[0].dim.worldSphere.radius = sJntSphInit.list->dim.modelSphere.radius * 0.03f * 100.0f;
+    } else {
+        this->actor.params = 0;
+        this->collider.list[0].dim.worldSphere.radius = sJntSphInit.list->dim.modelSphere.radius;
+    }
+
+    SkelAnime_ChangeAnimDefaultRepeat(&this->skelAnime, &D_060000F0);
+    Math_Vec3f_Copy(&this->actor.posRot, &this->actor.initPosRot);
+    this->actor.shape.rot.x = 0;
+    this->actor.shape.rot.z = 0;
+    this->timer = 300;
+    this->actor.shape.unk_08 = 2000;
+    this->actor.unk_4C = 2000.0f;
+    this->actor.draw = NULL;
+    this->actionFunc = func_809E10A8;
+}
+
+void EnCrow_Wait(EnCrow* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+    s32 skelanimeUpdated;
+    s16 var;
+
+    SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+    skelanimeUpdated = func_800A56C8(&this->skelAnime, 0.0f);
+    this->actor.speedXZ = (Math_Rand_ZeroOne() * 1.5f) + 3.0f;
+
+    if (this->actor.bgCheckFlags & 8) {
+        this->aimRotY = this->actor.wallPolyRot;
+    } else {
+        if (func_8002DBB0(&this->actor, &this->actor.initPosRot) > 300.0f) {
+            this->aimRotY = func_8002DAC0(&this->actor, &this->actor.initPosRot);
+        }
+    }
+
+    if (Math_SmoothScaleMaxMinS(&this->actor.shape.rot.y, this->aimRotY, 5, 0x300, 0x10) == 0 && skelanimeUpdated &&
+        Math_Rand_ZeroOne() < 0.1f) {
+
+        var = func_8002DAC0(&this->actor, &this->actor.initPosRot) - this->actor.shape.rot.y;
+        if (var > 0) {
+            this->aimRotY += 0x1000 + (0x1000 * Math_Rand_ZeroOne());
+        } else {
+            this->aimRotY -= 0x1000 + (0x1000 * Math_Rand_ZeroOne());
+        }
+
+        Audio_PlayActorSound2(&this->actor, NA_SE_EN_KAICHO_CRY);
+    }
+
+    if (this->actor.waterY > -40.0f) {
+        this->aimRotX = -0x1000;
+    } else {
+        if (this->actor.posRot.pos.y < this->actor.initPosRot.pos.y - 50.0f) {
+            this->aimRotX = -0x800 - (Math_Rand_ZeroOne() * 0x800);
+        } else if (this->actor.posRot.pos.y > this->actor.initPosRot.pos.y + 50.0f) {
+            this->aimRotX = 0x800 + (Math_Rand_ZeroOne() * 0x800);
+        }
+    }
+
+    if (Math_SmoothScaleMaxMinS(&this->actor.shape, this->aimRotX, 10, 0x100, 8) == 0 && skelanimeUpdated &&
+        Math_Rand_ZeroOne() < 0.1f) {
+        if (this->actor.initPosRot.pos.y < this->actor.posRot.pos.y) {
+            this->aimRotX -= (0x400 * Math_Rand_ZeroOne()) + 0x400;
+        } else {
+            this->aimRotX += (0x400 * Math_Rand_ZeroOne()) + 0x400;
+        }
+
+        this->aimRotX = CLAMP(this->aimRotX, -0x1000, 0x1000);
+    }
+
+    if (this->actor.bgCheckFlags & 1) {
+        Math_ApproxUpdateScaledS(&this->actor.shape, -0x100, 0x400);
+    }
+
+    DECR(this->timer);
```

Comment:
Also relevant in other functions:
```suggestion
    if (this->timer != 0) {
        this->timer--;
    }
```